app/scripts/controllers/permissions/specifications.js

```diff
@@ -33,58 +32,18 @@ export const PermissionNames = Object.freeze({
  * PermissionController.
  */
 export const CaveatFactories = Object.freeze({
-  [CaveatTypes.restrictReturnedAccounts]: (accounts) => {
-    return { type: CaveatTypes.restrictReturnedAccounts, value: accounts };
-  },
-
-  [CaveatTypes.restrictNetworkSwitching]: (chainIds) => {
-    return { type: CaveatTypes.restrictNetworkSwitching, value: chainIds };
-  },
+  [Caip25CaveatType]: createCaip25Caveat,
 });
 
 /**
  * Gets the specifications for all caveats that will be recognized by the
  * PermissionController.
- *
- * @param {{
- *   getInternalAccounts: () => Record<string, import('@metamask/keyring-api').InternalAccount>,
- * }} options - Options bag.
  */
-export const getCaveatSpecifications = ({
-  getInternalAccounts,
-  findNetworkClientIdByChainId,
-}) => {
+export const getCaveatSpecifications = () => {
   return {
-    [CaveatTypes.restrictReturnedAccounts]: {
-      type: CaveatTypes.restrictReturnedAccounts,
-
-      decorator: (method, caveat) => {
-        return async (args) => {
-          const result = await method(args);
-          return result.filter((account) => caveat.value.includes(account));
-        };
-      },
-
-      validator: (caveat, _origin, _target) =>
-        validateCaveatAccounts(caveat.value, getInternalAccounts),
-
-      merger: (leftValue, rightValue) => {
-        const newValue = Array.from(new Set([...leftValue, ...rightValue]));
-        const diff = newValue.filter((value) => !leftValue.includes(value));
-        return [newValue, diff];
-      },
-    },
-    [CaveatTypes.restrictNetworkSwitching]: {
-      type: CaveatTypes.restrictNetworkSwitching,
-      validator: (caveat, _origin, _target) =>
-        validateCaveatNetworks(caveat.value, findNetworkClientIdByChainId),
-      merger: (leftValue, rightValue) => {
-        const newValue = Array.from(new Set([...leftValue, ...rightValue]));
-        const diff = newValue.filter((value) => !leftValue.includes(value));
-        return [newValue, diff];
-      },
+    [Caip25CaveatType]: {
+      type: Caip25CaveatType,
```

Comment:
I see that we don't have a validator or mutator here. From the `CaveatSpecification` docs:

```
 * If no validator is specified, no validation of caveat values will be
 * performed. Although caveats can also be validated by permission validators,
 * validating caveat values separately is strongly recommended.
```

Nit: Thoughts on adding a validator? Otherwise any JSON value would be accepted. And while the caveat type we're using is _mostly_ validated in the permission validator (it looks like we forgot to validate the type of `sessionProperties`, which isn't used yet), the permission controller seems to expect the validation to occur in both places. Perhaps we could refactor the permission validator to move the caveat validation steps to a separate function.

Reply:
I assume the lack of a `mutator` is intentional (we aren't using `requestPermissionsIncremental` anymore). That seems OK. We can add that later when we need it.

Reply:
To clarify, is your suggestion that we move most of the logic currently in the CAIP-25 Permission [validator](https://github.com/MetaMask/core/blob/main/packages/multichain/src/caip25Permission.ts#L100) into a validator for the caveat specifically?

> the permission controller seems to expect the validation to occur in both places.
I don't think that it is currently the case in [main](https://github.com/MetaMask/metamask-extension/blob/2bd3db8994097460d535698ec9d9f0731f610610/app/scripts/controllers/permissions/specifications.js#L110) that validation of the caveat happens in two places. I see validation of the permission at a shallow level, and then validation of the caveat more thoroughly

Reply:
Yes exactly. Or keep it in both places? Perhaps @rekmarks could add further detail on why we might prefer this

Reply:
I would implement a caveat validator. As stated in the docs, you _can_ rely exclusively on the permission validator, but there are instances where the permission validator won't be called, specifically in the case of `updatePermissionsByCaveat()`. Even if we don't need it right now, it's easy to forget to add it later.

To make reasoning about caveat validation as simple as possible, I recommend the following separation of concerns:
- Let permission validators verify that a permission has the expected caveats
- Let caveat validators verify that a caveat has the expected shape / structure

Permissions and caveats should be designed such that this separation of concerns can be maintained.

> Aside: we call `validateModifiedPermission()` in `setCaveat()` but not `updatePermissionsByCaveat()`. I can't account for this discrepancy, but will look into it further.

Reply:
Update to aside: We do call `validateModifiedPermission()` as appropriate in `updatePermissionsByCaveat()`, via `deleteCaveat()`.

Reply:
https://github.com/MetaMask/core/pull/5064
https://github.com/MetaMask/metamask-extension/pull/29166